src/openms/source/FORMAT/Base64.cpp

```diff
@@ -1,123 +1,107 @@
 // Copyright (c) 2002-present, OpenMS Inc. -- EKU Tuebingen, ETH Zurich, and FU Berlin
 // SPDX-License-Identifier: BSD-3-Clause
-//
-// --------------------------------------------------------------------------
-// $Maintainer: Timo Sachsenberg $
-// $Authors: Marc Sturm, Chris Bielow, Moritz Aubermann $
-// --------------------------------------------------------------------------
+
+#pragma once
 
 #include <OpenMS/FORMAT/Base64.h>
+#include <OpenMS/SYSTEM/SIMDe.h>
 
 #include <QtCore/QList>
 #include <QtCore/QString>
-
-#include <OpenMS/SYSTEM/SIMDe.h>
+#include <OpenMS/FORMAT/Vec.h>
 
 using namespace std;
 
 namespace OpenMS
 {
-  const simde__m128i mask1_ = simde_mm_set1_epi32(0x3F000000); // 00111111 00000000 00000000 00000000
-  const simde__m128i mask2_ = simde_mm_set1_epi32(0x003F0000); // 00000000 00111111 00000000 00000000
-  const simde__m128i mask3_ = simde_mm_set1_epi32(0x00003F00); // 00000000 00000000 00111111 00000000
-  const simde__m128i mask4_ = simde_mm_set1_epi32(0x0000003F); // 00000000 00000000 00000000 00111111
+  const Vec128 mask1_ = simde_mm_set1_epi32(0x3F000000);
+  const Vec128 mask2_ = simde_mm_set1_epi32(0x003F0000);
+  const Vec128 mask3_ = simde_mm_set1_epi32(0x00003F00);
+  const Vec128 mask4_ = simde_mm_set1_epi32(0x0000003F);
+
+  const Vec128 mask1d_ = simde_mm_set1_epi32(0xFF000000);
+  const Vec128 mask2d_ = simde_mm_set1_epi32(0x00FF0000);
+  const Vec128 mask3d_ = simde_mm_set1_epi32(0x0000FF00);
+  const Vec128 mask4d_ = simde_mm_set1_epi32(0x000000FF);
+
+  const Vec128 difference_A_ = simde_mm_set1_epi8('A');
+  const Vec128 difference_a_ = simde_mm_set1_epi8('a' - 26);
+  const Vec128 difference_0_ = simde_mm_set1_epi8('0' - 52);
+  const Vec128 difference_plus_ = simde_mm_set1_epi8('+');
+  const Vec128 difference_slash_ = simde_mm_set1_epi8('/');
+
+  const Vec128 shuffle_mask_1_ = simde_mm_setr_epi8(2, 2, 1, 0, 5, 5, 4, 3, 8, 8, 7, 6, 11, 11, 10, 9);
+  const Vec128 shuffle_mask_2_ = simde_mm_setr_epi8(3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12);
+  const Vec128 shuffle_mask_big_endian_ = simde_mm_setr_epi8(0, 1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 8, 9, 10, 11, 11);
+  const Vec128 shuffle_mask_d_2_ = simde_mm_setr_epi8(3, 2, 1, 7, 6, 5, 11, 10, 9, 15, 14, 13, 0, 4, 8, 12);
+
+  void registerEncoder_(Vec128& data_raw)
+  {
+    Vec128 data(data_raw);
 
-  const simde__m128i mask1d_ = simde_mm_set1_epi32(0xFF000000); // 11111111 00000000 00000000 00000000
-  const simde__m128i mask2d_ = simde_mm_set1_epi32(0x00FF0000); // 00000000 11111111 00000000 00000000
-  const simde__m128i mask3d_ = simde_mm_set1_epi32(0x0000FF00); // 00000000 00000000 11111111 00000000
-  const simde__m128i mask4d_ = simde_mm_set1_epi32(0x000000FF); // 00000000 00000000 00000000 11111111
+    data = (!OPENMS_IS_BIG_ENDIAN ? data.shuffle_epi8(shuffle_mask_1_) : data.shuffle_epi8(shuffle_mask_big_endian_));
```

Comment:
using a ternary operator here is not recommended, since the code before used
`if constexpr ()`, which is guaranteed to be very efficient.
Please only make minimal (or rather no) changes to the code structure.